app-localize-behavior.html

```diff
@@ -210,20 +246,30 @@
 
       // Cache the key/value pairs for the same language, so that we don't
       // do extra work if we're just reusing strings across an application.
-      var messageKey = key + resources[language][key];
-      var msg = proto.__localizationCache.messages[messageKey];
+      var value = resources[language][key];
+      var msg;
+      if (value) {
+        var messageKey = key + value;
+        msg = proto.__localizationCache.messages[messageKey];
+      }
 
-      if (!msg) {
-        msg = new IntlMessageFormat(resources[language][key], language, formats);
-        proto.__localizationCache.messages[messageKey] = msg;
+      if (value && !msg) {
+        try {
+          msg = new IntlMessageFormat(value, language, formats);
+          if (msg) proto.__localizationCache.messages[messageKey] = msg;
+        } catch (e) {
+          if (!this.useKeyIfMissing) throw e;
```

Comment:
Does this throw? When does this throw? I think adding a comment about this would be nice.

Also, why not assign `msg` to `key` in this block, rather than doing another check in L270?